host_core/lib/host_core/host.ex

```diff
@@ -68,9 +68,95 @@ defmodule HostCore.Host do
     {:reply, state.host_key, state}
   end
 
-  defp start_gnat() do
-    {:ok, _gnat} = Gnat.start_link(%{host: '127.0.0.1', port: 4222}, name: :lattice_nats)
-    {:ok, _gnaT} = Gnat.start_link(%{host: '127.0.0.1', port: 4222}, name: :control_nats)
+  defp start_gnat(opts) do
+    configure_lattice_gnat(%{
+      host: opts.rpc_host,
+      port: opts.rpc_port,
+      username: opts.rpc_user,
+      password: opts.rpc_pass,
+      token: opts.rpc_token,
+      nkey_seed: opts.rpc_seed,
+      jwt: opts.rpc_jwt
+    })
+
+    configure_control_gnat(%{
+      host: opts.ctl_host,
+      port: opts.ctl_port,
+      username: opts.ctl_user,
+      password: opts.ctl_pass,
+      token: opts.ctl_token,
+      nkey_seed: opts.ctl_seed,
+      jwt: opts.ctl_jwt
+    })
+  end
+
+  defp configure_lattice_gnat(opts) do
+    conn_settings =
+      Map.merge(%{host: opts.host, port: opts.port}, determine_auth_method(opts, "lattice"))
+
+    case Gnat.start_link(conn_settings, name: :lattice_nats) do
+      {:ok, _gnat} ->
+        :ok
+
+      {:error, :econnrefused} ->
+        Logger.error("Unable to establish lattice NATS connection, connection refused")
+
+      {:error, _} ->
+        Logger.error("Authentication to lattice NATS connection failed")
```

Comment:
What's the return value of `Logger.error` ? Is it `:error` ? just trying to figure out what the return value of this function is when it fails and whether that corresponds to what `start_link`'s contract requires.

Reply:
re: https://hexdocs.pm/logger/1.12/Logger.html#error/2, it looks to return `:ok`, so it appears that `configure_lattice_gnat` will always return `:ok` unless an exception is thrown